### PR TITLE
Refactor query table action in Assistant Builder

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR is the final step in refactoring the UI for the query table action in the Assistant Builder. It allows us to use the same components consistently, whether we're working with documents or tables, providing a uniform experience for users.

Previously, tables operated differently, even though their functionality was similar to retrieval and process actions. A downside to the previous approach, which still persists, is that the table IDs returned by the server originate from core. This requires some gymnastic to integrate them with existing components. Ideally, we should create these IDs during generation, but addressing this is beyond the scope of this PR.

The new logic ensures that table queries use the same type as existing search-like actions, allowing us to eliminate many of the previously table-specific components and reduce the number of additional modals. I also had to refactor `buildInitialActions`, which impacts all actions, to maintain code readability and ease of maintenance.

There is a known issue with the new design: the hierarchy in the tree is not always set correctly. I've added a to-do note to address this in a future PR. Despite this, the current experience is still an improvement over what we had before.

## Risk

Pretty risky, could break the actions part of the Assistant Builder. Safe to rollback.

### CODE REVIEW

File: front/components/assistant_builder/server_side_props_helpers.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import type {
```
Comment (flvndvd) [inline front/components/assistant_builder/server_side_props_helpers.ts:1]: Mainly refactored into functions.

File: front/components/assistant_builder/server_side_props_helpers.ts [hunk 2]
```diff
@@ -69,209 +68,272 @@ export async function buildInitialActions({
...
+    (acc, curr) => ({
+      ...acc,
+      [curr.dataSourceView.sId]: curr,
+    }),
+    {} as DataSourceViewSelectionConfigurations
+  );
+}
 
-      processConfiguration.configuration.tagsFilter = action.tagsFilter;
+async function renderTableDataSourcesConfigurations(
```
Comment (flvndvd) [inline front/components/assistant_builder/server_side_props_helpers.ts:258]: This is the main change of the file.

File: front/components/assistant_builder/submitAssistantBuilderForm.ts [hunk 3]
```diff
@@ -120,7 +121,19 @@ export async function submitAssistantBuilderForm({
             type: "tables_query_configuration",
             name: a.name,
             description: a.description,
-            tables: Object.values(a.configuration),
+            tables: Object.values(a.configuration).flatMap(
+              ({ dataSourceView, selectedResources }) => {
+                return selectedResources.map((resource) => ({
+                  dataSourceId: dataSourceView.dataSource.name,
+                  dataSourceViewId: dataSourceView.sId,
+                  workspaceId: owner.sId,
```
Comment (Fraggle) [inline front/components/assistant_builder/submitAssistantBuilderForm.ts:129]: nit: Can we avoid using `owner` from the outside scope ?

**Review by Fraggle — APPROVED** [review]

File: front/components/assistant_builder/submitAssistantBuilderForm.ts (hunk 3 above)
Comment (flvndvd) [reply front/components/assistant_builder/submitAssistantBuilderForm.ts:129]: Given that we need the `sId` we can't really use something else.

File: front/components/assistant_builder/ActionsScreen.tsx [hunk 4]
```diff
@@ -650,7 +650,7 @@ function ActionConfigEditor({
               actionName: action.name,
               actionDescription: action.description,
               getNewActionConfig: (old) =>
-                setNewAction(old as AssistantBuilderTablesQueryConfiguration),
+                setNewAction(old as AssistantBuilderTableConfiguration),
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/components/assistant_builder/ActionsScreen.tsx:653]: It requires a bit more work to make it a templated function but happy to do it in a follow up PR.

**Pull request merged**